google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java

```diff
@@ -1323,7 +1323,8 @@ public void testSignUrlForBlobWithSpecialChars() throws NoSuchAlgorithmException
       String blobName = "/a" + specialChar + "b";
       URL url =
           storage.signUrl(BlobInfo.newBuilder(BUCKET_NAME1, blobName).build(), 14, TimeUnit.DAYS);
-      String escapedBlobName = UrlEscapers.urlPathSegmentEscaper().escape(blobName);
+      String escapedBlobName =
+          UrlEscapers.urlFragmentEscaper().escape(blobName).replace("?", "%3F");
```

Comment:
@lesv Escaping `&` is not necessary as long as we properly escape all `?`. See the improved integration test I pushed with 988c1c4

@mickeyreiss yours is a good refactoring suggestion that I plan on taking, however for the moment I say kick this out and fix the bug :D